### ISSUE TITLE
bedrock: L2 Constructs

### ISSUE THREAD

**Issue by jlosito** [aws/aws-cdk#29963]
### Describe the feature

I would like to use L2 constructs to provision Bedrock related resources with sensible defaults. Now that L1 constructs have been added via https://github.com/aws/aws-cdk/pull/29911 , this should be possible now!

### Use Case

I have been using the awslabs/generative-ai-cdk-constructs library to provision and configure Bedrock related resources, but this library makes use of custom resources to perform all of its actions. I much rather use native constructs rather than custom resources. 

https://github.com/awslabs/generative-ai-cdk-constructs 

### Proposed Solution

If possible, use the same defaults that the generative-ai-cdk-constructs library is using for each of its L2-like constructs. In my opinion, this would make it a little bit easier to users to migrate without having to change too much. If not, that's fine too.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.139.0

### Environment details (OS name and version, etc.)

macOS 14.4.1

**Comment by pahud** [comment]
Absolutely. Please help us prioritize with 👍 .

**Comment by hoegertn** [comment]
If somebody from the community wants to pick it up, that is exactly what https://www.open-constructs.org/ is for.

**Comment by badmintoncryer** [comment]
There is a Bedrock L2 construct at [awslabs/generative-ai-cdk-constructs](https://github.com/awslabs/generative-ai-cdk-constructs).

**Comment by jlosito** [comment]
@badmintoncryer those are all using custom resources. It would be better if they were using leveraging the L1 constructs instead that were just introduced within the last month.

**Comment by badmintoncryer** [comment]
@jlosito  It's true. That is using custom resources.
As you said, it seems good to create L2 using L1.

When I have time, I'll try to submit a PR to open-constructs.

**Comment by dirkmichelmycomosi** [comment]
Yes please, absolutely. Currently working with the L1s and the L2s would definitely help increase productivity.

**Comment by laithalsaadoon** [comment]
https://github.com/awslabs/generative-ai-cdk-constructs/pull/414 was merged today

**Comment by krokoko** [comment]
@pahud @paulhcsun we did look at the contribution guide and design guidelines to contribute to the CDK our existing L2 bedrock construct mentioned above (and maybe more). We will have some changes to do to follow the guidelines, in the meantime should we open a new GitHub issue to track this effort ? Is there anything else we should do to ? Thanks !

**Comment by krokoko** [comment]
Following the agents PR, submitting new PRs now for prompt management and guardrails:
- https://github.com/aws/aws-cdk/pull/34765
- https://github.com/aws/aws-cdk/pull/34754
